src/components/ConnectionsListProfiles.tsx

```diff
@@ -50,6 +69,21 @@ const ConnectionsListProfiles = ({
       ? Object.keys(following).map(profileForId)
       : [];
 
+  const changeGraphState = async (profile: Profile) => {
+    if (isFollowing(profile)) {
+      await unfollowUser(userId, profile.fromId);
+    } else {
+      await followUser(userId, profile.fromId);
+    }
+    dispatch(
+      updateRelationshipStatus({
+        follower: userId,
+        followee: profile.fromId,
+        status: RelationshipStatus.UPDATING,
+      })
+    );
+  };
+
   return (
     <>
       {connectionsList.map((userProfile) => (
```

Comment:
Not really part of this, but I found `userProfile` to be very confusing when trying to read the changes below. `connectionProfile` might be more clear in which profile we are referring?